### ISSUE TITLE
unable to start unpriviledged containers with shadow 4.4

### ISSUE THREAD

**Issue by ndk-** [lxc/lxc#1353]
lxc versions tested: 2.0.6, 2.0.5
shadow version: 4.4
OS: gentoo
kernel: 4.8.5
cgmanager is properly set up

Downgrading shadow to 4.2.1 without changing anything else solves the issue

Errors are standard in logs:
ERROR    lxc_conf - Failed to chown /dev/pts/5
ERROR    lxc_start - Failed to shift tty into container

**Comment by brauner** [comment]
Can you please append the log from a starting container with a shadow 4.4 and one from a starting container with a shadow 4.2.1? (Also, cgmanager is deprecated, gentoo should be updated to rely on `libpam-cgfs`. :))

**Comment by ndk-** [comment]
error:
[ub.log.txt](https://github.com/lxc/lxc/files/705777/ub.log.txt)

**Comment by hallyn** [comment]
Hi,

sorry, this pretty old - is this still happening?  (I assume so since you have a logfile attached).  If so can you please also attach a logfile with shadow 4.2.1?

When shadow 4.4 is installed, what does 'ls -l `which newuidmap`' show ?

**Comment by glggwp** [comment]
I have 
lxc: 2.0.7
kernel: 4.9.16
shadow: 4.4-r2 

After making the following amendments to conf.c ttys_shift_ids

```
3612     if (strcmp(c->console.name, "") !=0 && chown_mapped_root(c->console.name, c) < 0) {
3613         int en = errno;
3614         char *tt = strerror(en);
3615         ERROR("Failed to chown %s (%d => %s)", c->console.name, en, tt);
3616         return -1;
3617     }   
```
I get this [lxc-start_errno.txt](https://github.com/lxc/lxc/files/875308/lxc-start_errno.txt)

other files of interest:
unmodified lxc-start output [lxc-start.txt](https://github.com/lxc/lxc/files/875315/lxc-start.txt)
lxc-create output [lxc-create.txt](https://github.com/lxc/lxc/files/875317/lxc-create.txt)

**Comment by hallyn** [comment]
If instead you chmod u+s /usr/bin/new[ug]idmap , does that work right?

**Comment by glggwp** [comment]
Thanks, that's resolved my issue.

**Comment by hallyn** [comment]
Thanks for the update.

**Comment by linas** [comment]
The `chmod u+s /usr/bin/new[ug]idmap` fix implies that this is a dupe of bug #1454
